src/store/v3MigrationSelectors.ts

```diff
@@ -0,0 +1,472 @@
+) => {
+  const v3ReservesMap = userReservesData.reduce((obj, item) => {
+    obj[item.underlyingAsset] = item;
+    return obj;
+  }, {} as Record<string, ComputedUserReserve<ReserveDataHumanized & FormatReserveUSDResponse>>);
+
+  return v3ReservesMap;
+};
+
+export const selectUserReservesForMigration = (store: RootStore, timestamp: number) => {
+  const { userReservesData: userReserveV3Data, ...v3ReservesUserSummary } = selectV3UserSummary(
+    store,
+    timestamp
+  );
+
+  const { userReservesData: userReservesV2Data, ...v2ReservesUserSummary } =
+    selectUserSummaryAndIncentives(store, timestamp);
+
+  const poolReserveV3 = selectCurrentChainIdV3PoolReserve(store);
+  const userEmodeCategoryId = poolReserveV3?.userEmodeCategoryId;
+
+  let isolatedReserveV3 = selectIsolationModeForMigration(v3ReservesUserSummary);
+
+  const v3ReservesMap = selectUserReservesMapFromUserReserves(userReserveV3Data);
+
+  if (v3ReservesUserSummary.totalCollateralMarketReferenceCurrency == '0') {
+    const definitiveAssets = selectDefinitiveSupplyAssetForMigration(store, v3ReservesMap);
+    if (definitiveAssets.length > 0) {
+      const definitiveAsset = v3ReservesMap[definitiveAssets[0].underlyingAsset];
+      if (definitiveAsset.reserve.usageAsCollateralEnabled && definitiveAsset.reserve.isIsolated) {
+        isolatedReserveV3 = definitiveAsset.reserve;
+      }
+    }
+  }
+
+  const supplyReserves = userReservesV2Data.filter(
+    (userReserve) => userReserve.underlyingBalance !== '0'
+  );
+
+  const borrowReserves = userReservesV2Data.filter(
+    (reserve) => reserve.variableBorrows != '0' || reserve.stableBorrows != '0'
+  );
+
+  const mappedSupplyReserves = supplyReserves.map((userReserve) => {
+    let usageAsCollateralEnabledOnUser = true;
+    const isolatedOnV3 = v3ReservesMap[userReserve.underlyingAsset]?.reserve.isIsolated;
+    const canBeEnforced = v3ReservesMap[userReserve.underlyingAsset]?.underlyingBalance == '0';
+    if (isolatedReserveV3) {
+      usageAsCollateralEnabledOnUser =
+        userReserve.underlyingAsset == isolatedReserveV3.underlyingAsset;
+    } else {
+      const v3SupplyAsset = v3ReservesMap[userReserve.underlyingAsset];
+      if (v3SupplyAsset?.underlyingBalance !== '0') {
+        usageAsCollateralEnabledOnUser = v3SupplyAsset?.usageAsCollateralEnabledOnUser;
+      } else {
+        usageAsCollateralEnabledOnUser = !isolatedOnV3;
+      }
+    }
+    return {
+      ...userReserve,
+      usageAsCollateralEnabledOnUser,
+      isolatedOnV3,
+      canBeEnforced,
+    };
+  });
+
+  const mappedBorrowReserves = borrowReserves.map((userReserve) => {
+    // TOOD: make mapping for liquidity
+    let disabledForMigration = false;
+    const selectedReserve = v3ReservesMap[userReserve.underlyingAsset]?.reserve;
+
+    if (isolatedReserveV3) {
+      disabledForMigration = selectedReserve.borrowableInIsolation;
+    } else if (userEmodeCategoryId !== 0) {
```

Comment:
Currently this won't handle if a user is in eMode and isolationMode. To fix this you can replace the `else if` with `if(!disabledForMigration && userEmodeCategoryId !== 0) `

Reply:
fixed